presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SingleDistinctOptimizer.java

```diff
@@ -81,6 +81,9 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Optional<S
             if (masks.size() != 1 || node.getMasks().size() != node.getAggregations().size()) {
                 return context.defaultRewrite(node, Optional.empty());
             }
+            if (!Iterables.getOnlyElement(masks).getName().contains("distinct")) {
```

Comment:
This is because the SingleDistinctOptimizer remove the masks associated with the aggregation function and optimize it to a group by and it's unconditionally apply to all plan.  So with filter clause for single expression, it removes the mask.  The distinct mask was created with a hard coded "distinct" as the name.  The call site above this point has a list of optimizers passed in.   To do a check for every optimizer seems to be heavy.